src/category_theory/monoidal/preadditive.lean

```diff
@@ -47,4 +47,143 @@ local attribute [simp] monoidal_preadditive.tensor_add monoidal_preadditive.add_
+  end,
+  inv_hom_id' := begin
+    ext j j',
+    simp only [preadditive.sum_comp, preadditive.comp_sum,
+      category.assoc, category.comp_id, category.id_comp, dite_comp, comp_dite,
+      ←comp_tensor_id_assoc, biproduct.ι_π, biproduct.ι_π_assoc],
+    simp only [category.comp_id, category.id_comp, eq_to_hom_refl, tensor_dite,
+      finset.sum_dite_eq, finset.mem_univ, finset.sum_congr, finset.sum_dite_eq',
+      dite_eq_ite, if_t_t, if_true, dif_ctx_congr,
+      comp_zero, zero_comp, monoidal_preadditive.tensor_zero],
+    split_ifs with h,
+    { cases h, simp, },
+    { simp, },
+  end, }
+
+lemma right_distributor_assoc {J : Type*} [decidable_eq J] [fintype J] (X Y : C) (f : J → C) :
+   (right_distributor X f ⊗ as_iso (𝟙 Y)) ≪≫ right_distributor Y _ =
+     α_ (⨁ f) X Y ≪≫ right_distributor (X ⊗ Y) f ≪≫ biproduct.map_iso (λ j, (α_ _ X Y).symm) :=
+begin
+  ext,
+  simp only [category.comp_id, category.assoc, eq_to_hom_refl, iso.symm_hom,
+    iso.trans_hom, as_iso_hom, comp_zero, comp_dite, preadditive.sum_comp, preadditive.comp_sum,
+    sum_tensor, comp_tensor_id, tensor_iso_hom, right_distributor_hom,
+    biproduct.map_iso_hom, biproduct.ι_map, biproduct.ι_π,
+    finset.sum_dite_irrel, finset.sum_dite_eq', finset.sum_const_zero, finset.mem_univ, if_true],
+  simp only [←comp_tensor_id, biproduct.ι_π, dite_tensor, comp_dite],
+  simp only [category.comp_id, comp_tensor_id, eq_to_hom_refl, tensor_id, comp_zero,
+    monoidal_preadditive.zero_tensor,
+    if_true, dif_ctx_congr, finset.mem_univ, finset.sum_congr, finset.sum_dite_eq'],
+  simp only [←tensor_id, associator_inv_naturality, iso.hom_inv_id_assoc]
+end
+
+lemma left_distributor_right_distributor_assoc
+  {J : Type*} [decidable_eq J] [fintype J] (X Y : C) (f : J → C) :
+  (left_distributor X f ⊗ as_iso (𝟙 Y)) ≪≫ right_distributor Y _ =
+    α_ X (⨁ f) Y ≪≫ (as_iso (𝟙 X) ⊗ right_distributor Y _) ≪≫ left_distributor X _ ≪≫
+      biproduct.map_iso (λ j, (α_ _ _ _).symm) :=
+begin
+  ext,
+  simp only [category.comp_id, category.assoc, eq_to_hom_refl, iso.symm_hom,
+    iso.trans_hom, as_iso_hom, comp_zero, comp_dite, preadditive.sum_comp, preadditive.comp_sum,
+    sum_tensor, tensor_sum, comp_tensor_id, tensor_iso_hom,
+    left_distributor_hom, right_distributor_hom,
+    biproduct.map_iso_hom, biproduct.ι_map, biproduct.ι_π,
+    finset.sum_dite_irrel, finset.sum_dite_eq', finset.sum_const_zero, finset.mem_univ, if_true],
+  simp only [←comp_tensor_id, ←id_tensor_comp_assoc, category.assoc, biproduct.ι_π,
+    comp_dite, dite_comp, tensor_dite, dite_tensor],
+  simp only [category.comp_id, category.id_comp, category.assoc, id_tensor_comp,
+    comp_zero, zero_comp, monoidal_preadditive.tensor_zero, monoidal_preadditive.zero_tensor,
+    comp_tensor_id, eq_to_hom_refl, tensor_id,
+    if_true, dif_ctx_congr, finset.sum_congr, finset.mem_univ, finset.sum_dite_eq'],
+  simp only [associator_inv_naturality, iso.hom_inv_id_assoc]
+end
+
```

Comment:
Should this PR end with an instance of `preserves_finite_biproducts (X ⊗ _)`?

Reply:
Good suggestion. In fact, doing this first lets us avoid some yucky proofs. I don't just use the `@[simps]` lemma, and instead separately prove simp lemmas that the isomorphism coming from `preserves_finite_biproducts` actually have the explicit forms I'd previously given in the construction of the isomorphism.

I suspect if one thought really hard that the associativity lemmas I proved come from general nonsense about preserving bilimits, but the current approach (work in coordinates, and blast away with simp) is not terrible.